[TZone] Update to export annotation for exported WebCore classes

msaboff opened the pull request:
#### 98298c2126a1e78cc6e8f768a1f2076974864214
<pre>
[TZone] Update to export annotation for exported WebCore classes
<a href="https://bugs.webkit.org/show_bug.cgi?id=279924">https://bugs.webkit.org/show_bug.cgi?id=279924</a>
<a href="https://rdar.apple.com/136253091">rdar://136253091</a>

Reviewed by David Degazio.

Changed the TZone annotations for WEBCORE_EXPORT&apos;ed classes from WTF_MAKE_TZONE_ALLOCATED annotations to
WTF_MAKE_TZONE_ALLOCATED_EXPORT annotations and similar changes for the other flavors of *_TZONE_ALLOCATED
annotations.

Also fixed the #ifdef&apos;ing of WebKit/WebProcess/GPU/webrtc/MediaRecorderProvider.cpp to match the #ifdef&apos;ing
of Source/WebCore/Modules/mediarecorder/MediaRecorderProvider.cpp, its super class.

* Source/WebCore/Modules/applepay/ApplePayLogoSystemImage.h:
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDM.h:
* Source/WebCore/Modules/indexeddb/IDBDatabaseNameAndVersionRequest.h:
* Source/WebCore/Modules/indexeddb/client/IDBConnectionProxy.h:
* Source/WebCore/Modules/mediarecorder/MediaRecorderProvider.h:
* Source/WebCore/Modules/model-element/ModelPlayer.h:
* Source/WebCore/Modules/model-element/ModelPlayerProvider.h:
* Source/WebCore/Modules/reporting/Report.h:
* Source/WebCore/Modules/reporting/ReportBody.h:
* Source/WebCore/Modules/reporting/ReportingScope.h:
* Source/WebCore/Modules/speech/SpeechSynthesisUtterance.h:
* Source/WebCore/Modules/system-preview/ARKitBadgeSystemImage.h:
* Source/WebCore/Modules/webauthn/fido/FidoHidPacket.cpp:
* Source/WebCore/Modules/webauthn/fido/FidoHidPacket.h:
* Source/WebCore/dom/StaticNodeList.h:
* Source/WebCore/dom/TrustedHTML.h:
* Source/WebCore/dom/TrustedScript.h:
* Source/WebCore/dom/TrustedScriptURL.h:
* Source/WebCore/fileapi/AsyncFileStream.h:
* Source/WebCore/html/canvas/Path2D.h:
* Source/WebCore/html/track/WebVTTParser.h:
* Source/WebCore/inspector/InspectorFrontendClientLocal.h:
* Source/WebCore/platform/NowPlayingManager.h:
* Source/WebCore/platform/PlatformSpeechSynthesizer.h:
* Source/WebCore/platform/animation/AcceleratedEffectStack.h:
* Source/WebCore/platform/audio/AudioSession.h:
* Source/WebCore/platform/audio/cocoa/CAAudioStreamDescription.h:
* Source/WebCore/platform/audio/mac/SharedRoutingArbitrator.h:
* Source/WebCore/platform/cocoa/CoreLocationGeolocationProvider.h:
* Source/WebCore/platform/graphics/MIMETypeCache.h:
* Source/WebCore/platform/graphics/MediaPlayer.h:
* Source/WebCore/platform/graphics/PlatformTimeRanges.h:
* Source/WebCore/platform/graphics/avfoundation/objc/LocalSampleBufferDisplayLayer.h:
* Source/WebCore/platform/graphics/cg/ImageBufferIOSurfaceBackend.h:
* Source/WebCore/platform/graphics/mac/AppKitControlSystemImage.h:
* Source/WebCore/platform/graphics/mac/ScrollbarTrackCornerSystemImageMac.h:
* Source/WebCore/platform/ios/PlaybackSessionInterfaceAVKit.h:
* Source/WebCore/platform/ios/PlaybackSessionInterfaceIOS.h:
* Source/WebCore/platform/mac/PlaybackSessionInterfaceMac.h:
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateMock.h:
* Source/WebCore/platform/mediastream/AudioMediaStreamTrackRenderer.h:
* Source/WebCore/platform/mediastream/WebRTCProvider.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.h:
* Source/WebCore/platform/network/BlobRegistryImpl.h:
* Source/WebCore/platform/network/cocoa/CookieStorageObserver.h:
* Source/WebKit/WebProcess/GPU/webrtc/MediaRecorderProvider.cpp:

Canonical link: <a href="https://commits.webkit.org/283967@main">https://commits.webkit.org/283967@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a6d58e04d7d35359c17a96e9588368351820d927

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67697 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/47079 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/20336 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71751 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/18838 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/69815 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54878 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18644 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/54178 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/12586 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70764 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/43185 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/58543 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34646 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39859 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15948 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/17196 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61824 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/16290 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/73449 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11660 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15594 "Found 2 new test failures: fast/loader/subframe-navigate-during-main-frame-load.html http/tests/resourceLoadStatistics/no-third-party-cookie-blocking-when-itp-is-off.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/61630 "Found 2 new test failures: imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-with-classes-match-multiple-wildcard.html imported/w3c/web-platform-tests/pointerevents/coalesced_events_attributes_under_load.https.optional.html?mouse (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/11695 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58611 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61677 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15095 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9512 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/3132 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/42888 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43962 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/45149 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43701 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dhecht commented on Source/WebCore/Modules/applepay/ApplePayLogoSystemImage.h:
```diff
@@ -41,7 +41,7 @@ enum class ApplePayLogoStyle : bool {
 
 class WEBCORE_EXPORT ApplePayLogoSystemImage final : public SystemImage {
-    WTF_MAKE_TZONE_ALLOCATED(ApplePayLogoSystemImage);
+    WTF_MAKE_TZONE_ALLOCATED_EXPORT(ApplePayLogoSystemImage, WEBCORE_EXPORT);
```
The "class WEBCORE_EXPORT" decl doesn't cause all the methods/operators to be exported already?  Does this fix some linker error you're encountering?

msaboff replied:
For some of these classes I did get linking errors.  Given that these classes all have WEBCORE_EXPORT, the TZone annotations should match as other users of the WebCore framework may have dependencies on these exported classes.

dhecht replied:
Yeah, they could have dependencies, but I would have thought that the class-level exporting would cause everything in the class to be exported. (Notice how nothing else is explicitly exported in any of these classes which seems to imply the methods get exported, otherwise it seems there'd be nothing you can do with these classes). But maybe not. If it fixes the linker errors, then okay.

ddegazio reviewed (approved):
r=me

msaboff committed:
[TZone] Update to export annotation for exported WebCore classes
https://bugs.webkit.org/show_bug.cgi?id=279924
rdar://136253091

Reviewed by David Degazio.

Changed the TZone annotations for WEBCORE_EXPORT'ed classes from WTF_MAKE_TZONE_ALLOCATED annotations to
WTF_MAKE_TZONE_ALLOCATED_EXPORT annotations and similar changes for the other flavors of *_TZONE_ALLOCATED
annotations.

Also fixed the #ifdef'ing of WebKit/WebProcess/GPU/webrtc/MediaRecorderProvider.cpp to match the #ifdef'ing
of Source/WebCore/Modules/mediarecorder/MediaRecorderProvider.cpp, its super class.

* Source/WebCore/Modules/applepay/ApplePayLogoSystemImage.h:
* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDM.h:
* Source/WebCore/Modules/indexeddb/IDBDatabaseNameAndVersionRequest.h:
* Source/WebCore/Modules/indexeddb/client/IDBConnectionProxy.h:
* Source/WebCore/Modules/mediarecorder/MediaRecorderProvider.h:
* Source/WebCore/Modules/model-element/ModelPlayer.h:
* Source/WebCore/Modules/model-element/ModelPlayerProvider.h:
* Source/WebCore/Modules/reporting/Report.h:
* Source/WebCore/Modules/reporting/ReportBody.h:
* Source/WebCore/Modules/reporting/ReportingScope.h:
* Source/WebCore/Modules/speech/SpeechSynthesisUtterance.h:
* Source/WebCore/Modules/system-preview/ARKitBadgeSystemImage.h:
* Source/WebCore/Modules/webauthn/fido/FidoHidPacket.cpp:
* Source/WebCore/Modules/webauthn/fido/FidoHidPacket.h:
* Source/WebCore/dom/StaticNodeList.h:
* Source/WebCore/dom/TrustedHTML.h:
* Source/WebCore/dom/TrustedScript.h:
* Source/WebCore/dom/TrustedScriptURL.h:
* Source/WebCore/fileapi/AsyncFileStream.h:
* Source/WebCore/html/canvas/Path2D.h:
* Source/WebCore/html/track/WebVTTParser.h:
* Source/WebCore/inspector/InspectorFrontendClientLocal.h:
* Source/WebCore/platform/NowPlayingManager.h:
* Source/WebCore/platform/PlatformSpeechSynthesizer.h:
* Source/WebCore/platform/animation/AcceleratedEffectStack.h:
* Source/WebCore/platform/audio/AudioSession.h:
* Source/WebCore/platform/audio/cocoa/CAAudioStreamDescription.h:
* Source/WebCore/platform/audio/mac/SharedRoutingArbitrator.h:
* Source/WebCore/platform/cocoa/CoreLocationGeolocationProvider.h:
* Source/WebCore/platform/graphics/MIMETypeCache.h:
* Source/WebCore/platform/graphics/MediaPlayer.h:
* Source/WebCore/platform/graphics/PlatformTimeRanges.h:
* Source/WebCore/platform/graphics/avfoundation/objc/LocalSampleBufferDisplayLayer.h:
* Source/WebCore/platform/graphics/cg/ImageBufferIOSurfaceBackend.h:
* Source/WebCore/platform/graphics/mac/AppKitControlSystemImage.h:
* Source/WebCore/platform/graphics/mac/ScrollbarTrackCornerSystemImageMac.h:
* Source/WebCore/platform/ios/PlaybackSessionInterfaceAVKit.h:
* Source/WebCore/platform/ios/PlaybackSessionInterfaceIOS.h:
* Source/WebCore/platform/mac/PlaybackSessionInterfaceMac.h:
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateMock.h:
* Source/WebCore/platform/mediastream/AudioMediaStreamTrackRenderer.h:
* Source/WebCore/platform/mediastream/WebRTCProvider.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.h:
* Source/WebCore/platform/network/BlobRegistryImpl.h:
* Source/WebCore/platform/network/cocoa/CookieStorageObserver.h:
* Source/WebKit/WebProcess/GPU/webrtc/MediaRecorderProvider.cpp:

Canonical link: https://commits.webkit.org/283967@main
